src/__test__/index.test.ts

```diff
@@ -1,41 +1,272 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable import/first */
+// Some hackery to get WebSocket in the global namespace on nodejs
+// @todo: Find a nicer way to do this...
+;(global as any).WebSocket = require('isomorphic-ws')
+
 import { expect } from 'chai'
-import { NodeHttpTransport } from '@improbable-eng/grpc-web-node-http-transport/lib'
+import { NewStoreReply } from '@textile/threads-client-grpc/api_pb'
+import { WriteTransaction } from 'src/WriteTransaction'
 import { Client } from '../index'
 
-const client = new Client('http://localhost:9091', NodeHttpTransport())
+const sleep = (ms: number) => new Promise(r => setTimeout(r, ms))
+const client = new Client('http://localhost:9091')
 
-const personSchema = `{
-  "$id": "https://example.com/person.schema.json",
-  "$schema": "http://json-schema.org/draft-07/schema#",
-  "title": "Person",
-  "type": "object",
-  "properties": {
-    "firstName": {
-      "type": "string",
-      "description": "The person's first name."
+const personSchema = {
+  $id: 'https://example.com/person.schema.json',
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  title: 'Person',
+  type: 'object',
+  required: ['ID'],
+  properties: {
+    ID: {
+      type: 'string',
+      description: "The entity's id.",
+    },
+    firstName: {
+      type: 'string',
+      description: "The person's first name.",
+    },
+    lastName: {
+      type: 'string',
+      description: "The person's last name.",
     },
-    "lastName": {
-      "type": "string",
-      "description": "The person's last name."
+    age: {
+      description: 'Age in years which must be equal to or greater than zero.',
+      type: 'integer',
+      minimum: 0,
     },
-    "age": {
-      "description": "Age in years which must be equal to or greater than zero.",
-      "type": "integer",
-      "minimum": 0
-    }
+  },
+}
+
+interface Person {
+  ID: string
+  firstName: string
+  lastName: string
+  age: number
+}
+
+const createPerson = (): Person => {
+  return {
+    ID: '',
+    firstName: 'Adam',
+    lastName: 'Doe',
+    age: 21,
   }
 }
-`
 
-describe.skip('newStore', () => {
-  it('should resolve', async () => {
-    const store = await client.newStore()
-    expect(store.id).to.not.be.undefined
+describe('Client', function() {
```

Comment:
Most of these tests should match their GoLang counterparts. I have not included the `Find` tests in the PR. They will come in a separate PR because they require implementing the Query interface.